front/pages/w/[wId]/members/index.tsx

```diff
@@ -524,30 +517,78 @@ function InviteEmailModal({
         (e) => !members.find((m) => m.email === e)
       ),
     };
-    setInvitationsSegmentation(invitesByCase);
-    if (!shouldWarnAboutExistingMembers(invitesByCase)) {
+
+    const { notInWorkspace, activeDifferentRole, revoked } = invitesByCase;
+
+    const ReinviteUsersMessage = (
+      <div className="mt-6 flex flex-col gap-6 px-2">
+        {revoked.length > 0 && (
+          <div>
+            <div>
+              The user(s) below were previously revoked from your workspace.
+              Reinstating them will also immediately reinstate their
+              conversation history on Dust.
+            </div>
+            <div className="mt-2 flex max-h-48 flex-col gap-1 overflow-y-auto rounded border p-2 text-xs">
+              {revoked.map((user) => (
+                <div
+                  key={user.email}
+                >{`- ${user.fullName} (${user.email})`}</div>
+              ))}
+            </div>
+          </div>
+        )}
+        {activeDifferentRole.length > 0 && (
+          <div>
+            <div>
+              The user(s) below are already in your workspace with a different
+              role. Moving forward will change their role to{" "}
+              <span className="font-bold">{displayRole(invitationRole)}</span>.
+            </div>
+            <div className="mt-2 flex max-h-48 flex-col gap-1 overflow-y-auto rounded border p-2 text-xs">
+              {activeDifferentRole.map((user) => (
+                <div key={user.email}>{`- ${
+                  user.fullName
+                } (current role: ${displayRole(
+                  user.workspaces[0].role
+                )})`}</div>
+              ))}
+            </div>
+          </div>
+        )}
+
+        <div>Do you want to proceed?</div>
+      </div>
+    );
+
+    if (
+      !shouldWarnAboutExistingMembers(invitesByCase) ||
+      (await confirm({
+        title: "Some users are already in the workspace",
+        message: ReinviteUsersMessage,
+        validateLabel: "Yes, proceed",
+        validateVariant: "primaryWarning",
+      }))
+    ) {
+      await handleMembersRoleChange({
```

Comment:
There is an issue here, with the current implementation of `handleMembersRoleChange` you can't call it with an empty array, we will always try to mutate the first element:

https://github.com/dust-tt/dust/blob/2a004d473134462c2c814c016e1869d41d2297fa/front/pages/w/%5BwId%5D/members/index.tsx#L838

So either you refactor to use the workspace object, or handle empty object down the line but this function should not assume that `members` will always have at least one element.